pkg/models/dashboards.go

```diff
@@ -461,3 +462,24 @@ type GetDashboardRefByIdQuery struct {
 type UnprovisionDashboardCommand struct {
 	Id int64
 }
+
+type GetLibraryPanelsQuery struct {
+	DashboardId int64
+
+	Result map[string]LibraryPanel
+}
+
+type LibraryPanel struct {
+	ID       int64  `xorm:"pk autoincr 'id'"`
+	OrgID    int64  `xorm:"org_id"`
+	FolderID int64  `xorm:"folder_id"`
+	UID      string `xorm:"uid"`
+	Name     string
+	Model    json.RawMessage
+
+	Created time.Time
+	Updated time.Time
+
+	CreatedBy int64
+	UpdatedBy int64
+}
```

Comment:
This struct is already defined in the librarypanels package but when I tried to use that I got circular dependencies. Any ideas how to solve this?

Reply:
Can the dashboard api layer use the PanelLibraryService? you can add a new dependency here: https://github.com/grafana/grafana/blob/master/pkg/api/http_server.go#L81  

Then the api handler can just call the libraryService .GetLibararyPanelsForDashboard (or something like that)

Reply:
Yes, I saw there were already plenty of dependencies there, so I wasn't sure that was the correct way. I'll add the dependency to http server as suggested. Thank you!

Reply:
Doesn't seem to work though
```
Version: 7.4.0-pre, Linux Version: 7.4.0, Package Iteration: 1610632398pre
go build -ldflags -w -X main.version=7.4.0-pre -X main.commit=9fdb934535 -X main.buildstamp=1610621026 -X main.buildBranch=hugoh/dashboard-loadlibpanels -o ./bin/grafana-server ./pkg/cmd/grafana-server
package github.com/grafana/grafana/pkg/cmd/grafana-server
	imports github.com/grafana/grafana/pkg/server
	imports github.com/grafana/grafana/pkg/api
	imports github.com/grafana/grafana/pkg/services/librarypanels
	imports github.com/grafana/grafana/pkg/api: import cycle not allowed
```

Maybe the simpler micro service pattern doesn't work here, do you think I need to refactor the LibraryPanelService to the "old" style? Need to dig deeper and see if there is some examples already.

Reply:
interesting, hm.. might need to break out either http server from the root api package, or the api.Wrap & util functions into a separate api sub package. 

If dashboard has it's own service (that handled http api & db) then this would not be a problem, this looks like it's only a problem due to api package having dashboards api handlers & the base api interfaces & util functions & types that all "micro" services use.  

thoughts @aknuds1  @marefr on how to solve this? (handler in api package being able to reference a service that defines api handlers itself)

Reply:
I moved `common.go` into a `utils` folder. Naming here is tricky I tried `common`, `helpers`, `apiHelpers` but stuck with `utils`. After that I moved structs and functions into the LibraryPanelService instead. I highly appreciate any feedback about the naming and the move, thank you 🙏

Reply:
Hmm on second thought I think I'll create a separate PR with that move.

Reply:
> Hmm on second thought I think I'll create a separate PR with that move.
:+1:

Reply:
Opened https://github.com/grafana/grafana/pull/30298 instead